sdk/src/types/block/semantic.rs

```diff
@@ -368,6 +396,16 @@ pub fn semantic_validation(
         return Ok(Some(TransactionFailureReason::SumInputsOutputsAmountMismatch));
     }
 
+    if context.input_mana > context.output_mana
+        && context
+            .essence
+            .capabilities()
+            .has_capability(TransactionCapabilityFlag::BurnMana)
```

Comment:
Shouldn't this be the opposite?
```suggestion
        && !context
            .essence
            .capabilities()
            .has_capability(TransactionCapabilityFlag::BurnMana)
```